src/Parser/Druid/Feral/Modules/FeralCore/Snapshot.js

```diff
@@ -0,0 +1,127 @@
+import SPELLS from 'common/SPELLS';
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import { encodeTargetString } from 'Parser/Core/Modules/EnemyInstances';
+
+const debug = false;
+
+/*
+Feral has a snapshotting mechanic which means the effect of some buffs are maintained over the duration of
+some DoTs even after the buff has worn off.
+Players should follow a number of rules with regards when they refresh a DoT and when they do not, depending
+on what buffs the DoT has snapshot and what buffs are currently active.
+
+RAKE snapshots:
+  Prowl, Tiger's Fury, Bloodtalons
+RIP snapshots:
+  Tiger's Fury, Bloodtalons
+MOONFIRE_FERAL snapshots:
+  Tiger's Fury
+THRASH_FERAL snapshots: (but isn't used in single target situations)
+  Tiger's Fury, Moment of Clarity
+
+The Snapshot class is 'abstract', and shouldn't be directly instantiated. Instead classes should extend
+it and test how well the combatant is making use of the snapshot mechanic.
+*/
+
+const TIGERS_FURY_MULTIPLIER = 1.15;
+const PROWL_MULTIPLIER = 2.00; // also applied by Incarnation: King of the Jungle, and Shadowmeld
+const BLOODTALONS_MULTIPLIER = 1.20;
+
+const JAGGED_WOUNDS_MODIFIER = 0.80;  // "[...]deal the same damage as normal but in 20% less time."
+const PANDEMIC_FRACTION = 0.3;
+
+// leeway in ms after loss of bloodtalons/prowl buff to count a cast as being buffed. Keep well below GCD to avoid false positives.
+const BUFF_WINDOW_TIME = 300;
```

Comment:
Is it common that the delay is that high? Otherwise I'd suggest lowering it to 50 or 100ms, as you can still get false positives from the buffs expirering naturally.

Reply:
Good point on the expiring buffs causing false positives. Especially for Prowl and Shadowmeld which can be lost through taking damage, which is a real risk on some fights.
Highest delay I could find on a log was 36ms, so I set the window to a hopefully safe 60ms.